app/policies/wiki_policy.rb

```diff
@@ -2,4 +2,30 @@ class WikiPolicy < ApplicationPolicy
   def show?
     (user.present? && user.admin?) || !record.private? || (record.private? && record.user_id == user.id)
   end
+  def permitted_attributes
+    if user.admin? || user.premium?
+      [:title, :body, :private]
+    else
+      [:title, :body]
+    end
+  end
+
+  class Scope
+    attr_reader :user, :scope
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def resolve
+      if user && user.admin?
+        scope.all
+      elsif user && user.premium?
+        t = Wiki.arel_table
+        scope.where(t[:private].eq(false).or(t[:user_id].eq(user.id)))
```

Comment:
I tried about 10 different ways and couldn't get this to work correctly without using arel. Could you suggest how to go about this without it?